extensions/interactions/ItemSelectionInput/ItemSelectionInput_test.py

```diff
@@ -0,0 +1,219 @@
+                'explanation': {
+                    'content_id': 'solution',
+                    'html': 'This is <i>solution</i> for state1'
+                }
+            },
+            'id': 'ItemSelectionInput'
+        }
+        item_selection_input_instance = (
+            interaction_registry.Registry.get_interaction_by_id(
+                'ItemSelectionInput'))
+        interaction_domain = (
+            state_domain.InteractionInstance.from_dict(
+                interaction_dict))
+        item_selection_proto = (
+            item_selection_input_instance.to_android_item_selection_input_proto(
+                interaction_domain.default_outcome,
+                interaction_domain.customization_args,
+                interaction_domain.hints,
+                interaction_domain.answer_groups))
+
+        item_customization_args = (
+            item_selection_proto.customization_args)
+        self.assertEqual(
+            item_customization_args.max_allowable_selection_count, 2)
+        self.assertEqual(
+            item_customization_args.min_allowable_selection_count, 1)
+        self.assertEqual(
+            item_customization_args.choices[0].content_id,
+            'ca_choices_2')
+        self.assertEqual(
+            item_customization_args.choices[1].content_id,
+            'ca_choices_3')
+        self.assertEqual(
+            item_customization_args.choices[0].text,
+            '<p>Choice 1</p>')
+        self.assertEqual(
+            item_customization_args.choices[1].text,
+            '<p>Choice 2</p>')
+
+        item_default_outcome = item_selection_proto.default_outcome
+        self.assertEqual(
+            item_default_outcome.destination_state, 'abc')
+        self.assertEqual(
+            item_default_outcome.feedback.text, 'Correct Answer')
+        self.assertEqual(
+            item_default_outcome.feedback.content_id, 'feedback_1')
+        self.assertTrue(item_default_outcome.labelled_as_correct)
+
+        self.assertEqual(
+            item_selection_proto.hints[0].hint_content.content_id, 'hint_1')
+        self.assertEqual(
+            item_selection_proto.hints[1].hint_content.content_id, 'hint_2')
+        self.assertEqual(
+            item_selection_proto.hints[0].hint_content.text,
+            '<p>This is a first hint.</p>')
+        self.assertEqual(
+            item_selection_proto.hints[1].hint_content.text,
+            '<p>This is the second hint.</p>')
+
+        item_answer_group = (
+            item_selection_proto.answer_groups[0]
+                .base_answer_group.outcome)
+        self.assertEqual(
+            item_answer_group.destination_state, 'abc')
+        self.assertTrue(item_answer_group.labelled_as_correct)
+        self.assertEqual(
+            item_answer_group.feedback.content_id,
+            'feedback_1')
+        self.assertEqual(
+            item_answer_group.feedback.text,
+            '<p>Feedback</p>')
+
+        item_rule_spec = (
```

Comment:
Here and elsewhere: item --> item_selection.

In general, use the full names for things. Please fix the other files as well, if necessary.

Reply:
updated everywhere in this PR.